test/unit/test_model_server.py

```diff
@@ -0,0 +1,330 @@
+        "--model-store",
+        torchserve.MODEL_STORE,
+        "--ts-config",
+        torchserve.TS_CONFIG_FILE,
+        "--log-config",
+        torchserve.DEFAULT_TS_LOG_FILE,
+        "--models",
+        "model.mar"
+    ]
+
+    subprocess_popen.assert_called_once_with(ts_model_server_cmd)
+    sigterm.assert_called_once_with(retrieve.return_value)
+
+
+@patch("subprocess.call")
+@patch("subprocess.Popen")
+@patch("sagemaker_pytorch_serving_container.torchserve._retrieve_ts_server_process")
+@patch("sagemaker_pytorch_serving_container.torchserve._add_sigterm_handler")
+@patch("sagemaker_pytorch_serving_container.torchserve._create_torchserve_config_file")
+@patch("sagemaker_pytorch_serving_container.torchserve._adapt_to_ts_format")
+def test_start_torchserve_custom_handler_service(
+    adapt, create_config, sigterm, retrieve, subprocess_popen, subprocess_call
+):
+    handler_service = Mock()
+
+    torchserve.start_torchserve(handler_service)
+
+    adapt.assert_called_once_with(handler_service)
+
+
+@patch("sagemaker_pytorch_serving_container.torchserve._set_python_path")
+@patch("subprocess.check_call")
+@patch("os.makedirs")
+@patch("os.path.exists", return_value=False)
+def test_adapt_to_ts_format(path_exists, make_dir, subprocess_check_call, set_python_path):
+    handler_service = Mock()
+
+    torchserve._adapt_to_ts_format(handler_service)
+
+    path_exists.assert_called_once_with(torchserve.DEFAULT_TS_MODEL_DIRECTORY)
+    make_dir.assert_called_once_with(torchserve.DEFAULT_TS_MODEL_DIRECTORY)
+
+    model_archiver_cmd = [
+        "torch-model-archiver",
+        "--model-name",
+        torchserve.DEFAULT_TS_MODEL_NAME,
+        "--handler",
+        handler_service,
+        "--serialized-file",
+        os.path.join(environment.model_dir, torchserve.DEFAULT_TS_MODEL_SERIALIZED_FILE),
+        "--export-path",
+        torchserve.DEFAULT_TS_MODEL_DIRECTORY,
+        "--extra-files",
+        os.path.join(environment.model_dir, environment.Environment().module_name + ".py"),
+        "--version",
+        "1",
+    ]
+
+    subprocess_check_call.assert_called_once_with(model_archiver_cmd)
+    set_python_path.assert_called_once_with()
+
+
+@patch("sagemaker_pytorch_serving_container.torchserve._set_python_path")
+@patch("subprocess.check_call")
+@patch("os.makedirs")
+@patch("os.path.exists", return_value=True)
+def test_adapt_to_ts_format_existing_path(
+    path_exists, make_dir, subprocess_check_call, set_python_path
+):
+    handler_service = Mock()
+
+    torchserve._adapt_to_ts_format(handler_service)
+
+    path_exists.assert_called_once_with(torchserve.DEFAULT_TS_MODEL_DIRECTORY)
+    make_dir.assert_not_called()
+
+
+@patch.dict(os.environ, {torchserve.PYTHON_PATH_ENV: PYTHON_PATH}, clear=True)
+def test_set_existing_python_path():
+    torchserve._set_python_path()
+
+    code_dir_path = "{}:".format(environment.code_dir)
+
+    assert os.environ[torchserve.PYTHON_PATH_ENV] == code_dir_path + PYTHON_PATH
```

Comment:
same here about preferring format strings to concatenation

```suggestion
    code_dir_path = "{}:{}".format(environment.code_dir, PYTHON_PATH)

    assert os.environ[torchserve.PYTHON_PATH_ENV] == code_dir_path
```